plugins/cindy3d/src/js/PrimitiveRenderer.js

```diff
@@ -219,9 +229,22 @@ PrimitiveRenderer.prototype.addPrimitive = function(attributes) {
     throw new GlError("Wrong number of attributes given");
   if (this.count == this.capacity) {
     let c = this.capacity*2, nd, nda, ndi, i, j, k, o, e = this.elements;
+    if (c > this.maxCount) {
+      if (this.capacity < this.maxCount) {
+        c = this.maxCount;
+      } else {
+        this.indexByteCount = 4;
+        this.itemTotalByteCount = this.itemAttribByteCount +
+          this.numElements * this.indexByteCount;
+        this.maxCount = Math.floor((1 << 31) / this.itemAttribByteCount);
```

Comment:
`1 << 31` seems dangerous to me as it will be interpreted as signed, thus negative, number. But Anyway, we are sure that `this.itemAttribByteCound` is sufficiently large :)
Actually, for uint32 we would like to have something like `(1<<32-1)` (which is different from -1 in javascript) but I guess this is more hypothetical as these numbers are rather beyond our scope and `this.maxCount` will never be used after that.